src/browser/vdiff.js

```diff
@@ -19,6 +19,13 @@ mocha.setup({
 });
 /* eslint-enable */
 
+function findRealSize(elem) {
+	if (!elem.shadowRoot) return elem;
+	const nestedSizedElem = elem.shadowRoot.querySelector('.vdiff-size');
+	if (!nestedSizedElem) return elem;
+	return findRealSize(nestedSizedElem);
```

Comment:
Cool. So if an element doesn't define its size at all but instead that falls to something in its shadow root, it can just put the class on that element and it'll recurse -- nice!

Reply:
Yeah exactly! And in theory, this should "just work" for consumers wrapping a `core` component in that situation, because we would've had to already add it for everything that needs it for our own tests.